src/com/facebook/buck/jvm/java/JavaBinary.java

```diff
@@ -242,14 +242,7 @@ public boolean isCacheable() {
 
   @Override
   public Stream<BuildTarget> getRuntimeDeps(BuildRuleResolver buildRuleResolver) {
-    Stream<BuildTarget> transitiveRuntimeDeps = Stream.of();
-    for (JavaLibrary lib : getTransitiveClasspathDeps()) {
-      // Skip ourself to avoid infinite recursion.
-      if (lib == this) continue;
-
-      transitiveRuntimeDeps =
-          Stream.concat(transitiveRuntimeDeps, lib.getRuntimeDeps(buildRuleResolver));
-    }
-    return transitiveRuntimeDeps;
+    return getTransitiveClasspathDeps().stream()
+        .flatMap(lib -> lib.getRuntimeDeps(buildRuleResolver));
```

Comment:
We probably still need to have a check that the lib is == this and return an empty stream in that case

Reply:
That was actually an impossible check - `getTransitiveClasspathDeps` is a list of `JavaLibrary`s, but `this` is a `JavaBinary` - they don't share any common ancestors so it's impossible to have that equality check ever be true.